pkg/resources/secret_with_oauth_authorization_code_grant_acceptance_test.go

```diff
@@ -314,12 +314,73 @@ func TestAcc_SecretWithAuthorizationCodeGrant_ExternalChange(t *testing.T) {
 				},
 				Check: resource.ComposeTestCheckFunc(
 					assert.AssertThat(t,
-						resourceassert.SecretWithAuthorizationCodeResource(t, secretModel.ResourceReference()).
+						resourceassert.SecretWithAuthorizationCodeGrantResource(t, secretModel.ResourceReference()).
 							HasOauthRefreshTokenExpiryTimeString(refreshTokenExpiryDateTime),
 						assert.Check(resource.TestCheckResourceAttrSet(secretModel.ResourceReference(), "describe_output.0.oauth_refresh_token_expiry_time")),
 					),
 				),
 			},
+			{
+				RefreshState: true,
```

Comment:
I see this kind of test case for the first time, could you briefly describe what it does/tests ?

Reply:
When discussing the `refresh_token_expiry_time` field with @sfc-gh-jmichalak we wanted to check if there will be a perma diff or not when invoking Read on that resource in order to be sure if we needed a diff suppress.

There was no perma diff because this filed is not being overwritten by value in describe, it is handled by using `setStateToValuesFromConfig`

`withExternalChangesMarking` and `handleExternalValueChangesToObjectInDescribe` is being used only to detect the external change on that field.

Reply:
Diffs after read are usually implicitly caught by the testing framework, but If we want to check that explicitly let's make a comment above this test case to know what are we actually asserting and what do we expect to happen.